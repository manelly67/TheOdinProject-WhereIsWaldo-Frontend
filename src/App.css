body {
  color: var(--color-base-text);
  background-color: var(--color-bg-first);
}
 

#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
  color: var(--color-base-text);
  background-color: var(--color-bg-first);
}

.setOfButtons {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  gap: 10px;
}

.iconImg {
  height: 30px;
  width: auto;
}


@media (max-width: 770px) {
 .bar {
  position: absolute;
  top: 4%;
  width: 300px;
  display: grid;
  grid-template-rows: 40px;
  grid-template-columns: 1fr 1fr 1fr 1fr 2fr;

 }
 .themeButton{
  grid-row: 1;
  grid-column-start: 5;
  grid-column-end: 6;
 }

.setOfLinks{
  grid-row: 1;
  grid-column-start: 1;
  grid-column-end:3;
  }

}


.setOfLinks {
  position: absolute;
  top: 15%;
  right: 3%;

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  gap: 3px;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: var(--color-bg-button);
  cursor: pointer;
  transition: border-color 0.25s;
  transition: transform 250ms ease-in-out;
}
button:hover {
  border-color: #84cc16;
  transform: scale(1.02);
}
button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}


.themeButton {
  position: absolute;
  top: 4%;
  right: 3%;
  z-index: 1;
  padding: 0.3em 0.9em;

  transition: transform 250ms ease-in-out;
}

.themeButton:hover {
  transform: scale(1.1);
 
}

.themeButton:active {
  transform: scale(1.1);
 
}
