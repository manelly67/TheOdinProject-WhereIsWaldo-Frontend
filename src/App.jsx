import { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import "./styles/App.css";
import ToggleTheme from "./components/ToggleTheme";

let didInit = false;

function App() {
  const titleDiv = document.querySelector("title");
  if (titleDiv) {
    titleDiv.textContent = "WHERE IS WALDO";
  }

  const navigate = useNavigate();
  const name_game_1 = "Waldo In The Galactic City";
  const name_game_2 = "Oh! Waldo is not here";
  const [playerId, setPlayerId] = useState(null);
  const [playerName, setPlayerName] = useState(null);
  const player = playerId === null ? null : { id: playerId, name: playerName };

  useEffect(() => {
    if (!didInit) {
      didInit = true;
      // âœ… Only runs once per app load
      const id = crypto.randomUUID();
      setPlayerId(id);
      setPlayerName("anonymous");
    }
  }, []);

  console.log(player);

  return (
    <>
      <ToggleTheme theme="light" />
      <h1>Where is Waldo - The Game</h1>

      <div className="setOfButtons">
        <button
          onClick={() => {
            navigate("/board", {
              replace: true,
              state: { player: player, gameName: name_game_1 },
            });
          }}
        >
          <h3>{name_game_1}</h3>
        </button>
        <button
          onClick={() => {
            navigate("/board", {
              replace: true,
              state: { player: player, gameName: name_game_2 },
            });
          }}
        >
          <h3>{name_game_2}</h3>
        </button>
      </div>
    </>
  );
}

export default App;
