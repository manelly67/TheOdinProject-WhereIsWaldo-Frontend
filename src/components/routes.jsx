import App from "../App";
import ErrorPage from "./Error_page";
import DrawingBoard from "./DrawingBoard";
import TopTen from "./TopTen";

const routes = [
  {
    index: true,
    path: "/",
    element: <App />,

    errorElement: <ErrorPage />,
  },
  {
    path: "board",
    element: <DrawingBoard />,
  },
  {
    path: "top_ten",
    element: <TopTen />,
  },
];

export default routes;
