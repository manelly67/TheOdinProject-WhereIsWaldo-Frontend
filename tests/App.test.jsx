import {
  MemoryRouter,
  createBrowserRouter,
  RouterProvider,
} from "react-router-dom";

import { describe, it, expect } from "vitest";
import { render, screen, waitFor, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import toHaveStyle from "@testing-library/jest-dom";
import mockdata from "../src/mock_data.jsx";

import App from "../src/App.jsx";
import routes from "../src/components/routes.jsx";

describe("App Component", () => {
  it("display the page", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();

    expect(screen.getAllByRole("button").length).toBe(1);
  });

  it("should display fetched data", async () => {
    // a rather simple mock, you might use something more advanced for your needs
    const mock_getPlayer = vi.fn();
    mock_getPlayer({
      message: 'new player created',
      player: {
        id: '11c10925-dd64-4cf3-96f0-c7d805b092a3',
        playername: 'ANONIMOUS',
        sessionId: 'XreW60dmse2ueeS5MNU5cB4ISxiPM9Xe',
        Game: []
      }
    });
    const mock_update = vi.fn();

    let resolve;
    function fetch() {
      return new Promise((_resolve) => {
        resolve = _resolve;
      });
    }

    const {rerender} =  render(
    <MemoryRouter initialEntries={["/"]}>
          <App
            getPlayer={mock_getPlayer}
            updatePlayerObj={mock_update}
            playerId={resolve}
          />
        </MemoryRouter>
      );

    await act(async () => {
     [resolve] = await mock_getPlayer.mock.calls[0];
    });
    console.log(resolve); 
    rerender(
      <MemoryRouter initialEntries={["/"]}>
        <App
          getPlayer={mock_getPlayer}
          updatePlayerObj={mock_update}
          playerId={resolve}
        />
      </MemoryRouter>
    );
  });
  // eL RERENDER NO ESTA FUNCIONANDO Y NO ACTUALIZA EL DOM
});

/* ESTE TEST CAMBIARLO CUANDO SE HAYA TRAIDO EL USUARIO DE LA PAGINA WEB
 
 it("navigate to drawing board", async () => {
    const router = createBrowserRouter(routes);
    render(<RouterProvider router={router} />);

    const user = userEvent.setup();

    expect(screen.getByText(/Waldo In The Galactic City/i)).toBeInTheDocument();
    const buttons = screen.getAllByRole("button");
    const [,second,third] = buttons;

    await user.click(second);

    expect(
      screen.getByAltText(/Waldo In The Galactic City/i)
    ).toBeInTheDocument();

    const homeLink = screen.getByText(/home/i);
    // back to home
    await user.click(homeLink);
    // select other image
    await user.click(third);

    waitFor(()=>{
      expect(
        screen.getByAltText(/Oh! Waldo is not here/i)
      ).toBeInTheDocument();
  
    });

  });
}); */
