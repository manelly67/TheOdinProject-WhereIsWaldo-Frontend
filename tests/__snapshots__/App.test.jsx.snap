// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > display the page 1`] = `
<div>
  <button
    class="themeButton"
  >
    <img
      alt="theme-light-dark"
      class="iconImg"
      height="30px"
      src="/src/assets/theme-light-dark.png"
      width="30px"
    />
  </button>
  <h1>
    Where is Waldo - The Game
  </h1>
  <div>
    <p>
      loading...
    </p>
    <hr />
    <p>
      Dear user, this is a study project, please wait 1 minute for the server to wake up.
    </p>
  </div>
</div>
`;
